neurom/io/datawrapper.py

```diff
@@ -168,31 +177,30 @@ def _extract_sections(data_block):
             curr_section.ids.append(row_id)
 
         if row_id in sec_end_pts:
-            parent_section[curr_section.ids[-1]] = len(_sections) - 1
-            _sections.append(Section())
-            curr_section = _sections[-1]
+            parent_section[curr_section.ids[-1]] = len(sections) - 1
             # Parent-child discontinuity section
             if gap:
+                sections.append(DataBlockSection())
```

Comment:
this is where the problem is: an extra `DataBlockSection` was being created unnecessarily for the last row if it was a leaf